### PR TITLE
update roadmap

### PR DESCRIPTION
**Pull request by teto**

### CODE REVIEW

File: roadmap.html [hunk 1]
```diff
@@ -36,9 +36,12 @@ <h3><a href="https://github.com/neovim/neovim/milestone/19">0.5</a></h3>
   <li>Extended marks (text properties, decorations, virtual text)</li>
 </ul>
 
-<h3>0.6</h3>
+<h3><a href="https://github.com/neovim/neovim/milestone/11">0.6</a></h3>
 <ul>
   <li>Better file-change detection</li>
+  <li>TUI as a remote UI</li>
+  <li>Virtual lines</li>
+  <li>Better folds</li>
```
Comment (teto) [inline roadmap.html:44]: diagnostic API ?

Comment (bfredl) [reply roadmap.html:44]: yes.

File: roadmap.html [hunk 2]
```diff
@@ -36,9 +36,13 @@ <h3><a href="https://github.com/neovim/neovim/milestone/19">0.5</a></h3>
   <li>Extended marks (text properties, decorations, virtual text)</li>
```
Comment (justinmk) [inline roadmap.html:36]: i don't think 0.5 has Lua remote plugin host, so that should be moved to 0.6 also?

File: roadmap.html [hunk 3]
```diff
@@ -36,9 +36,13 @@ <h3><a href="https://github.com/neovim/neovim/milestone/19">0.5</a></h3>
...
 </ul>
 
-<h3>0.6</h3>
+<h3><a href="https://github.com/neovim/neovim/milestone/11">0.6</a></h3>
 <ul>
   <li>Better file-change detection</li>
+  <li>Terminal UI as a remote UI</li>
+  <li>Virtual lines (think virtualtext but on new lines)</li>
+  <li>Improved folds</li>
+  <li>Diagnostic API</li>
```
Comment (justinmk) [inline roadmap.html:45]: assuming this is for LSP, or is a general UI subsystem enhancement?

```suggestion
  <li>LSP: Diagnostic API</li>
```

Comment (bfredl) [reply roadmap.html:45]: I think it is meant to be generic, i e reuse the LSP UI logic even for non-LSP sources

Comment (clason) [reply roadmap.html:45]: Yes, the motivation was for people to (be able to) stop shoehorning random linters into LSP via null-ls or efm while still getting decent-looking diagnostics (that don't fight each other).

**Comment by clason** [comment]
superseded by #243

**Pull request closed without merging**